### PR TITLE
Fix keystone v3

### PR DESCRIPTION
**Pull request by tbachman**
The keystone v3 integration was broken. This adds an additional
section to the configuration file, apic_ml2_auth, which allows
the specification of credential information needed by apicapi,
so that it can look up the openstack projects in keystone. This
fix intentionally mirrors the implementation in GBP, so that any
changes can be kept consistent between the two repos.

The old style of configuration is also kept, in order to preserve
backwards-compatibility.

### CODE REVIEW

File: apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# project owning each neutron resource, which by default, requires
+# admin. If we keep this, we should probably move it to a separate
+# config module. But we should also investigate whether admin is even
+# needed, or if neutron's credentials could somehow be used.
+AUTH_GROUP = 'apic_aim_auth'
+ksc_session.Session.register_conf_options(cfg.CONF, AUTH_GROUP)
+ksc_auth.register_conf_options(cfg.CONF, AUTH_GROUP)
+
+
+class ProjectNameCache(object):
```
Comment (amittbose) [inline apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py]: do we need this whole class when the Ml2 driver won't use most of it? Why not just copy the code for _get_keystone_client?

File: apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from neutron._i18n import _LW
+
+LOG = logging.getLogger(__name__)
+
+# REVISIT: We use keystone to get the name of the keystone
+# project owning each neutron resource, which by default, requires
+# admin. If we keep this, we should probably move it to a separate
+# config module. But we should also investigate whether admin is even
+# needed, or if neutron's credentials could somehow be used.
+AUTH_GROUP = 'apic_aim_auth'
```
Comment (amittbose) [inline apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py]: I think there should not be references to 'aim' in this code at all. Also the commit message mentions the name of this section to "apic_ml2_auth" but the code uses "apic_aim_auth"

**Review by amittbose — CHANGES_REQUESTED** [review]

File: apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py (hunk 1 above)
Comment (tbachman) [reply apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py]: Will remove.

File: apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py (hunk 2 above)
Comment (tbachman) [reply apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py]: Mandeep and I discussed this. The naming is unfortunate, as we've already had a release with this in our release notes (i.e. we thought we were already using this for v3 keystone config across all variants -- it ends up the legacy apic-ml2-driver wasn't using it). The decision was to keep the same name in order to keep this consistent.

Comment (tbachman) [reply apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py]: Will fix apic_ml2_auth -- my original patch changed it to that name, before Mandeep and I had discussed it.

**Comment by tbachman** [comment]
Pushed an update, addressing reviewer comments.

**Review by amittbose — APPROVED** [review]

**Pull request merged**